src/heretic/main.py

```diff
@@ -388,20 +401,31 @@ def objective_wrapper(trial: Trial) -> tuple[float, float]:
 
     best_trials = sorted(
         study.best_trials,
-        key=lambda trial: trial.user_attrs["refusals"],
+        key=lambda trial: (
+            trial.user_attrs["refusals"],
+            max(trial.user_attrs["kl_divergence"], 0),
```

Comment:
KL divergences cannot be negative by construction. Did you encounter negative values? If so, that would indicate a serious bug in KLD computation.

Reply:
Agreed. This can go, iirc was a safeguard/overkill ai suggestion that I kept in